src/directLine.ts

```diff
@@ -272,15 +275,31 @@ export class DirectLine implements IBotConnection {
     constructor(options: DirectLineOptions) {
         this.secret = options.secret;
         this.token = options.secret || options.token;
+        this.webSocket = typeof WebSocket !== 'undefined' && WebSocket !== undefined; 
+
         if (options.domain)
             this.domain = options.domain;
         if (options.webSocket !== undefined)
-            this.webSocket = options.webSocket;
-
+            this.webSocket = this.webSocket && options.webSocket;
+        if (options.conversationId) {
+            this.conversationId = options.conversationId;
+        }
+        if (options.watermark !== undefined) {
```

Comment:
I kept it like that because watermark can to be equal to 0. This said it is passed as a string so I will change that.